tests/test_optionals.py

```diff
@@ -61,3 +63,19 @@ def test_help_with_shorts(capture_output):
 
     assert err.value.code == 0
     assert "-v --value-1" in capture_output.captured_output
+
+
+class OptionalValueOldTyping(BaseModel):
+    my_first_optional: Optional[int] = None
+
+
+def test_optional_value_old_typing(capture_output):
+    capture_output(OptionalValueOldTyping, [])
```

Comment:
Can you change these tests for them to have assertions (and explicit testing) ? Like so:

```python
class Model(BaseModel):
    value: int | None = None

def test_optional_with_none():
    model = parse(Model, ["--value", "10"])
    assert model == Model(value=10)

class OldModel(BaseModel):
    value: Optional[int] = None

def test_old_model_with_none():
    model = parse(OldModel, ["--value", "10"])
    assert model == OldModel(value=10)
```